test/e2e/kubectl.go

```diff
@@ -792,3 +825,36 @@ func getUDData(jpgExpected string, ns string) func(*client.Client, string) error
 		}
 	}
 }
+
```

Comment:
given we need an os.File anyway, just implement the blocking reader like this:

```
// newBlockingReader returns a reader that allows reading the given string,
// then blocks until Close() is called on the returned closer
func newBlockingReader(s string) (io.Reader, io.Closer, error) {
  r, w, err := os.Pipe()
  if err != nil {
    return nil, nil, err
  }
  w.Write([]byte(s))
  return r, w
}
```

and use it like 
```
r, c, err := newBlockingReader()
if err != nil {
  Failf(...)
}
defer c.Close()
...
```